runtime/src/system_instruction_processor.rs

```diff
@@ -483,6 +483,28 @@ pub fn process_instruction(
                 instruction_context.try_borrow_instruction_account(transaction_context, 0)?;
             authorize_nonce_account(&mut me, &nonce_authority, &signers, invoke_context)
         }
+        SystemInstruction::UpgradeNonceAccount => {
+            let separate_nonce_from_blockhash = invoke_context
+                .feature_set
+                .is_active(&feature_set::separate_nonce_from_blockhash::id());
+            if !separate_nonce_from_blockhash {
+                return Err(InstructionError::InvalidInstructionData);
+            }
+            instruction_context.check_number_of_instruction_accounts(1)?;
```

Comment:
`check_number_of_instruction_accounts` is not necessary here as `UpgradeNonceAccount` is added as a whole. It was only used for older instructions to keep the error priorities the same without a feature gate.

Reply:
Though keeping it is consistent with the rest of the instruction processing and might make folks wonder why other places and not here.  I would vote to keep it here for consistency with the other instructions and remove them all at once when deemed appropriate.